subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/resolution_rules.adoc

```diff
@@ -183,6 +183,81 @@ include::sample[dir="snippets/dependencyManagement/customizingResolution-conditi
 Note that a project that is substituted must be included in the multi-project build (via `settings.gradle`).
 Dependency substitution rules take care of replacing the module dependency with the project dependency, but do not implicitly include the project in the build.
 
+[[sec:variant_aware_substitutions]]
+=== Substituting a dependency with another variant
+
+Gradle's dependency management engine is <<variant_model.adoc#understanding-variant-selection,variant-aware>> meaning that for a single component, the engine may select different artifacts and transitive dependencies.
+
+What to select is determined by the attributes of the consumer configuration and the attributes of the variants found on the producer side.
+It is, however, possible that some specific dependencies override attributes from the configuration itself.
+This is typically the case when using the <<java_platform_plugin.adoc#java_platform_plugin,Java Platform plugin>>: this plugin builds a special kind of component which is called a "platform" and can be addressed by setting the component category attribute to "platform", in opposition to typical dependencies which are targetting libraries.
```

Comment:
I would use code font (with backticks) for the values of the category attribute.